pkg/v3/tickers/recovery.go

```diff
@@ -1,19 +1,28 @@
 package tickers
 
 import (
+	"sync"
 	"time"
 
 	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
 )
 
 type recoveryTicker struct {
 	*retryTicker
-	// lastBlock ocr2keepers.BlockKey
+	lock      sync.RWMutex
+	lastBlock ocr2keepers.BlockKey
 }
 
-// getterFn is a function that retrieves the recoveryTick for the given time.
+// setBlock updates the block of the given payload
+func (rt *recoveryTicker) setBlock(p ocr2keepers.UpkeepPayload) {
+	rt.lock.Lock()
+	defer rt.lock.Unlock()
```

Comment:
I think this can be:

```
rt.lock.RLock()
defer rt.lock.RUnlock()
```

Since the `rt` is being read from, not written to?

Reply:
This is correct because this function should write to the `latestBlock`

Reply:
Ah okay, so the logic on line 20 should be flipped?